pkg/gadgets/internal/histogram/histogram.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2023 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package histogram provides a Histogram struct that represents a histogram of
+// the number of events that occurred in each interval. The intervals are powers
+// of two, so the slots array is indexed by the power of two minus one. For
+// example, the first slot is the number of events that occurred in the interval
+// [1, 1], the second slot is [2, 3], the third slot is [4, 7], and so on.
+
+package histogram
+
+import (
+	"fmt"
+	"strings"
+)
+
+type Slot struct {
+	Count         uint64 `json:"count"`
+	IntervalStart uint64 `json:"intervalStart"`
+	IntervalEnd   uint64 `json:"intervalEnd"`
+}
+
+// Histogram represents a histogram of the number of events that occurred in
+// each interval.
+type Histogram struct {
+	Unit  string `json:"unit,omitempty"`
+	Slots []Slot `json:"slots,omitempty"`
+}
```

Comment:
I considered defining a `Unit` type with its possible values like this:

```suggestion
type Unit string

const (
	UnitMilliseconds Unit = "msecs"
	UnitMicroseconds Unit = "usecs"
)

// Histogram represents a histogram of the number of events that occurred in
// each interval.
type Histogram struct {
	Unit  Unit   `json:"unit,omitempty"`
	Slots []Slot `json:"slots,omitempty"`
}
```

It will simplify the caller's life by allowing this:

```golang
h := histogram.NewFromCSlots(histogram.UnitMicroseconds, []uint32{1}),
```

Instead of:

```golang
h := histogram.NewFromCSlots("usecs", []uint32{1}),
```

However, it'll require exporting this package in `pkg` to make it usable also by integration tests. WDYT?

cc @eiffel-fl @mauriciovasquezbernal

Reply:
> I was wandering about the unit but so far it seems to only be printed. I mean, if we change the unit it does not have any impact on what is drawn (i.e. the underlying computation will not be done in another unit than µs), right?
>
The profile block-io gadget only supports microseconds, so this change is just a matter of using a variable instead of a literal string. Please notice that the bpf code is able to provide info in micro and milliseconds, but we are not using such a flag for this gadget.

> It could nonetheless be a good idea to introduce this new type as it would ease adding a PR to make the whole configurable if we want to use other time.
>
Exactly. I added support for micro and milliseconds in the profile tcprtt gadget that I'm working so this new type would help (also when we will add the same support to profile block-io).

> Nonetheless, rather than declaring our own type, would it be possible to use something from time golang package?
> 
I didn't find anything. The time package uses another notation: `µs`, and don't use constants, check [this](https://cs.opensource.google/go/go/+/refs/tags/go1.20.3:src/time/time.go;l=644). Do you know any package that provides it?